### ISSUE TITLE
State of the project

### ISSUE THREAD

**Issue by Chris2011** [Zwaen91/vuejs-netbeans#1]
Hi,

I'm interested to know what the currect state of the project is. What do you plan for it etc.

Regards

Chris

**Comment by Chris2011** [comment]
Is the project still alive?

**Comment by Zwaen91** [comment]
Hi @Chris2011 ,

at the moment the project is not continued s my MacBook is not able to handle the requirements of NetBeans Plugin Development.

Sorry.

**Comment by Chris2011** [comment]
Let us talk in slack or via mail, maybe I can help you, if you have problems: https://netbeans.signup.team/ or my mail christian[dot]lenz[at]gmx[dot]net.


Regards

Chris

**Comment by Chris2011** [comment]
Or is it a hardware problem?

**Comment by tlf30** [comment]
@Zwaen91, I am interested in building a vue plugin for netbeans and am wondering if it would be ok if I use this project as a starting point.   
  
Thanks,  
~Trevor

**Comment by Zwaen91** [comment]
Hello @tlf30 (Trevor),
feel free to use it as starting point. But I have to mention that I was at the beginning and was struggling with finding information and having problem with the platform. Better you make your own experiences.
Best,
Sven

**Comment by tlf30** [comment]
@Zwaen91 thank you! I have worked on netbeans plugins before thankfully. I agree entirely that the documentation is very poor, which it too bad because I love the netbeans platform. I will keep you up to date on my progress. 

~Trevor

**Comment by Chris2011** [comment]
@tlf30 hey Trevor, it would be good, if you can contribute to this project instead of creating a new one. Let us talk in slack together, to be more detailed and about the problem we had. I helped him (only a bit, not much) and we have one big problem: https://issues.apache.org/jira/browse/NETBEANS-88.

There is no need to have 2 different vuejs plugins, IMHO.


Cheers

Chris

**Comment by Chris2011** [comment]
Let us talk in slack together :)

**Comment by tlf30** [comment]
@Chris2011, I have branched mine off of this one, so it will be easy to create a pr and merge it in. But only if @Zwaen91 wants this repo to be the master copy. I have scrapped most of the existing code as it is. This was just a good template for getting started. 

I am on the netbeans slack channel, you can direct message me. I am `Trevor Flynn`

**Comment by Chris2011** [comment]
Great. I will talk to you soon :) maybe a group chat with @Zwaen91 too :)

**Comment by chiccomask** [comment]
Does anyone carried on with the netbeans plugin for vue.js?

**Comment by Chris2011** [comment]
Not really atm. The problem is we need to highlight each section, which is supported in a vue file separately. The workaround of using the mimetype text/vue+html is not supported. Please see those tickets: https://issues.apache.org/jira/browse/NETBEANS-504 (Vue support for NetBeans) and https://issues.apache.org/jira/browse/NETBEANS-88 (Problem with text/vue+html). So atm this is unfortunately to say. Here.

**Comment by Chris2011** [comment]
No one atm is really care about the HTML5/JS section and the other thing is, Vue depents on NBTS (NetBeans TypeScript) when you use 'lang="ts"'. So there is also work todo...

**Comment by chiccomask** [comment]
actually, enabling HTML5/JS in NetBeans and associating .vue extension with text/xhtml works well,  at least for code colouring.

**Comment by Chris2011** [comment]
Yes, this is the only one, what you can do as a workaround of using html, js and css. But not using Jade/Pug, TypeScript or LESS/SASS/SCSS or stylus. And you have no icon.

**Comment by patoessy** [comment]
Hello @chris, I have seen the vue specification in apache netbeans. Whats the state now? Can we have .nbms just for testing only.

**Comment by Chris2011** [comment]
Hey @patoessy, Thx for your interest :).

TL:DR;
I think it is not possible to add VueJS to NetBeans as a 3rd-Party-Plugin, because of not public functionality of the tokens of the languages that are needed for vue (HTML, Jade, JS, CSS, etc.) This is just a guess but back in the days, I tried of using tokens of those languages to add hints and I can't access them from the outside, except my package is a friend dependency of the used package, which is not nice at all.

So for now, just the Spec is there and an umbrella ticket with some of the discussion. That's it for now. I don't have enough knowledge to impelement it at all. I can try to use the Textmate lexer, but I think it is also not possible, because vetur (The official VS Code plugin for Vue) has for each section a textmate file with HTML as the parent. And I think parent languages (as a dependency inside of a textmate file) is not implemented out of the box atm.


So there is no progress at all yet, unfortunately, sry.


Cheers

Chris
